Source/Hypermedia.Client/Resolver/ILinkHcoCache.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using Bluehands.Hypermedia.Client.Hypermedia;
+
+namespace Bluehands.Hypermedia.Client.Resolver
+{
+    public interface ILinkHcoCache<TIdentifier>
+    {
+        bool TryGetValue(Uri uri, out CacheEntry<TIdentifier> entry);
+
+        void Set(Uri uri, CacheEntry<TIdentifier> entry);
+
+        void Remove(Uri uri);
+
+        void Clear();
+
+    }
+    public class CacheEntry<TIdentifier>
```

Comment:
```suggestion
    public readonly struct CacheEntry<TIdentifier>
```
Struct statt Class um unnötige Allocation eines Container-Objects zu sparen (CacheEntry wird erzeugt, übergeben und direkt entpackt/verwertet)